### ISSUE TITLE
Don't error if I don't have a requirements.txt

### ISSUE THREAD

**Issue by nealrichardson** [posit-dev/publisher#2205]
I had a new project I was starting. I had the code but hadn't created a requirements.txt file. I figured I'd click "Deploy" anyway and see what happens. It errored: 

![image](https://github.com/user-attachments/assets/cea49ea8-0848-4ccd-8a38-5c7b0b6f10c5)

What would be better is to pop open the "Python packages" section that shows the "Scan" button and invite me to scan. Don't show me an error.

**Comment by fh-mthomson** [comment]
I came across this while trying to deploy a `.ipynb` notebook as a "static doc" from VSCode (local) -> Connect via Publisher.

I'd done this previously in Jupyter Hub (in Workbench)-> Connect, and, I believe, it had auto-generated a placeholder `requirements.txt` before bundling and deploying to Connect.

Is it possible to deploy via Publisher _after_ rendering _without_ a `requirements.txt` file with the current Publisher?

**Comment by dotNomad** [comment]
> Is it possible to deploy via Publisher _after_ rendering _without_ a `requirements.txt` file with the current Publisher?

It is not possible today unfortunately. Right now Publisher does not auto-generate package files like `requirements.txt` during the deploy process.

Our "Python Packages" view in the sidebar does have a handy "Scan" button to help create a `requirements.txt` for you:

![Image](https://github.com/user-attachments/assets/20bba588-fdcc-4118-a4f8-14bb572330dc)

Or you could use `pip freeze > requirements.txt` (or another package manager equivalent) to create a `requirements.txt` to deploy with using Publisher.

**Comment by zackverham** [comment]
@dotNomad @christierney is this one covered by the work you just recently landed?

**Comment by christierney** [comment]
No, we have improved behavior when you have a lock file or toml with deps in it, but if you truly have no dependency tracking at all, the behavior is the same. Agreed it would be better to present the scan option directly (or auto-scan?) rather than show an error.

**Comment by dotNomad** [comment]
We could show the scan button in the error toast, as an option
